pkg/ast/ast_operation_definition.go

```diff
@@ -16,6 +17,22 @@ const (
 	OperationTypeSubscription
 )
 
+// Name returns a human-readable operation name for the given OperationType.
+// If the operation is not one of the OperationType constants, it panics.
+func (t OperationType) Name() string {
+	switch t {
+	case OperationTypeUnknown:
+		return "unknown"
+	case OperationTypeQuery:
+		return "query"
+	case OperationTypeMutation:
+		return "mutation"
+	case OperationTypeSubscription:
+		return "subscription"
+	}
+	panic(fmt.Errorf("unknown operation type %d", int(t)))
```

Comment:
Let's return an empty string. You already tracking this as an error.

Reply:
An empty string is a bad idea. If it gets propagated to other places then it's very difficult to track where the unknown value came from. We can either panic or return a string that's easy to search for.

Reply:
ok